hs_access_control/models/user.py

```diff
@@ -1368,11 +1371,12 @@ def can_change_resource(self, this_resource):
 
         return False
 
-    def can_change_resource_flags(self, this_resource):
+    def can_change_resource_flags(self, this_resource, check_shareable=False):
         """
         Whether self can change resource flags.
 
         :param this_resource: Resource to check.
+        :param  check_shareable: if True only check for the shareable flag
```

Comment:
I don't understand with this is labeled check_shareable.  The check it toggles is for published.

Reply:
I agree.  There's a split personality problem here. Either a user can change resource flags and the application determines which one, or we have a check by resource flag. The weird part is that we only check for a specific condition for one flag.  I will think about this.

Reply:
Reverted changes to this access control function. Added a new function to check if user can change shareable flag `can_change_resource_shareable_flag()`. 
@alvacouch For policy related to setting published resource flags refer to issue #2559